File: Electronics_Shop_17.Services/Database/Camera.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class Camera : IHelperId
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        //public virtual Product Product { get; set; }

        // Technical Specifications
        public decimal Megapixels { get; set; } // Sensor resolution
        public string SensorType { get; set; } // e.g., Full-frame, APS-C, Micro Four Thirds
        public string LensMount { get; set; } // e.g., Canon RF, Nikon Z
        public string VideoResolution { get; set; } // e.g., "4K at 60fps"

        // Physical Characteristics
        public decimal Weight_grams { get; set; } // Weight in grams
        public string Dimensions_mm { get; set; } // e.g., "138 x 97 x 88 mm"

        // Connectivity
        public bool HasWiFi { get; set; }
        public bool HasBluetooth { get; set; }

        // Battery
        public string BatteryType { get; set; } // e.g., "LP-E6NH"
        public int BatteryLife { get; set; } // Number of shots or recording time in minutes
    }
}

File: flutter17_mobile/lib/models/camera.dart
import 'package:flutter17_mobile/models/product.dart';
import 'package:json_annotation/json_annotation.dart';

part 'camera.g.dart';

@JsonSerializable()
class Camera {
  int? id;
  int? productId;

  // Technical Specifications
  double? megapixels;
  String? sensorType;
  String? lensMount;
  String? videoResolution;

  // Physical Characteristics
  double? weight;
  String? dimensions;

  // Connectivity
  bool? hasWiFi;
  bool? hasBluetooth;

  // Battery
  String? batteryType;
  int? batteryLife;

  Camera({
    this.id,
    this.productId,
    this.megapixels,
    this.sensorType,
    this.lensMount,
    this.videoResolution,
    this.weight,
    this.dimensions,
    this.hasWiFi,
    this.hasBluetooth,
    this.batteryType,
    this.batteryLife,
  });

  factory Camera.fromJson(Map<String, dynamic> json) => _$CameraFromJson(json);

  Map<String, dynamic> toJson() => _$CameraToJson(this);
}
